components/bottomSheets/SourceSelectSheet.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {
+  RefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  BottomSheetModal,
+  BottomSheetView,
+  BottomSheetBackdrop,
+} from '@gorhom/bottom-sheet';
+import { View, StyleSheet, Platform } from 'react-native';
+import { useTheme } from '../../context/ThemeContext';
+import { Theme } from '../../styles/colors';
+import { Source } from '../../database/sourcesRepository';
+import { useSourceStore } from '../../store/sourceStore';
+import { useChatStore } from '../../store/chatStore';
+import { useSQLiteContext } from 'expo-sqlite';
+import { useLLMStore } from '../../store/llmStore';
+import { useSourceUpload } from '../../hooks/useSourceUpload';
+import ActiveSourcesSection from './source-select/ActiveSourcesSection';
+import SourceListSection from './source-select/SourceListSection';
+import EmptySourcesView from './source-select/EmptySourcesView';
+
+interface Props {
+  bottomSheetModalRef: RefObject<BottomSheetModal | null>;
+  enabledSources: number[];
+  chatId: number | null;
+}
+
+const SourceSelectSheet = ({
+  bottomSheetModalRef,
+  enabledSources,
+  chatId,
+}: Props) => {
+  const { theme } = useTheme();
+  const { sources } = useSourceStore();
+  const { enableSource } = useChatStore();
+  const { sendEventMessage } = useLLMStore();
+  const db = useSQLiteContext();
+  const { uploadSource } = useSourceUpload();
+  const styles = useMemo(() => createStyles(theme), [theme]);
+  const [search, setSearch] = useState('');
+  const [activeSources, setActiveSources] = useState<Source[]>([]);
+  const [deactivatedSources, setDeactivatedSources] = useState<Source[]>([]);
```

Comment:
`enableSource` should already update the store where `enabledSources` are coming from. Do we need the local copy of these two lists for something else?